sage/vector_store.py

```diff
@@ -133,19 +134,23 @@ def upsert_batch(self, vectors: List[Vector], namespace: str):
         self.index.upsert(vectors=pinecone_vectors, namespace=namespace)
 
     def as_retriever(self, top_k: int, embeddings: Embeddings, namespace: str):
-        if self.bm25_encoder:
-            return PineconeHybridSearchRetriever(
-                embeddings=embeddings,
-                sparse_encoder=self.bm25_encoder,
-                index=self.index,
-                namespace=namespace,
-                top_k=top_k,
-                alpha=self.alpha,
-            )
-
-        return LangChainPinecone.from_existing_index(
+        bm25_retriever = BM25Retriever(
+            embeddings=embeddings,
+            sparse_encoder=self.bm25_encoder,
+            index=self.index,
+            namespace=namespace,
+            top_k=top_k,
+        ) if self.bm25_encoder else None
+
+        dense_retriever = LangChainPinecone.from_existing_index(
             index_name=self.index_name, embedding=embeddings, namespace=namespace
         ).as_retriever(search_kwargs={"k": top_k})
+        
+        retrievers = [bm25_retriever, dense_retriever] if bm25_retriever else [dense_retriever]
+        if len(retrievers) == 1:
+            return retrievers[0]
+        return EnsembleRetriever(retrievers=retrievers, weights=[1.0 / len(retrievers)] * len(retrievers))
```

Comment:
Could you use self.alpha instead of equal weights so that we can weigh dense vs sparse?